src/learner.cc

```diff
@@ -494,8 +494,17 @@ class LearnerConfiguration : public Learner {
           << std::numeric_limits<unsigned>::max() << " features or greater";
       num_feature = std::max(num_feature, static_cast<uint32_t>(num_col));
     }
-    // run allreduce on num_feature to find the maximum value
-    rabit::Allreduce<rabit::op::Max>(&num_feature, 1, nullptr, nullptr, "num_feature");
+
+    try {
+      // try run allreduce on num_feature to find the maximum value
+      rabit::Allreduce<rabit::op::Max>(&num_feature, 1, nullptr, nullptr,
+                                       "num_feature");
+    } catch (...) {
```

Comment:
- It's `dmlc::Error`.
- There's a variable in rabit called `STOP_PROCESS_ON_ERROR`, with `true` as default value.  Which means rabit calls `exit(1)` on failure.  See `rabit/include/rabit/internal/utils.h::HandleCheckError`.

Reply:
Ah it doesn't seem practical to recover in that case. I'm out of ideas. If we leave the all reduce call, all calls to configuration (save/load/serialisation) must be synchronous among workers. I've already seen a number of failures on dask due to this.

Reply:
One suggestion is don't copy the booster on dask training.

Reply:
Then you will get memory usage problems, also there might be other situations where configure gets called in distributed environments when saving or loading.